### ISSUE TITLE
Issue Using ERA5 Forcing

### ISSUE THREAD

**Issue by SorooshMani-NOAA** [schism-dev/pyschism#30]
When I try to use ERA5 as sflux1 I get an error in the `nws.nws2.nws2.NWS2.write` [method](https://github.com/schism-dev/pyschism/blob/ec00dffb251e49759a1162eff4e911c82a7d0cf2/pyschism/forcing/nws/nws2/nws2.py#L64). Looking at the code it seems that ERA5 is not following the same API as other sflux subclasses for `write` method. For example GFS [accepts](https://github.com/schism-dev/pyschism/blob/ec00dffb251e49759a1162eff4e911c82a7d0cf2/pyschism/forcing/nws/nws2/gfs.py#L325) a `level` argument while ERA5 [does not](https://github.com/schism-dev/pyschism/blob/ec00dffb251e49759a1162eff4e911c82a7d0cf2/pyschism/forcing/nws/nws2/era5.py#L290). This results in ERA5's `write` to get `1` as positional argument for `start_date` and causes issue due to passing `start_date` as keyword arg.

**Comment by cuill** [comment]
ERA5 was not added to NWS2 class, which writes data twice, first writing to tmpdir then rewriting as sflux_*.nc. Instead, ERA5 only writes once. The following script will generate sflux_1 from ERA5.

from datetime import datetime
from time import time
import pathlib

from pyschism.forcing.nws.nws2.era5 import ERA5

from pyschism.mesh.hgrid import Hgrid

startdate=datetime(2017, 4, 26) 
rnday=180

t0=time()
hgrid=Hgrid.open('./hgrid.gr3',crs='EPSG:4326')
bbox = hgrid.get_bbox('EPSG:4326', output_type='bbox')

er=ERA5()
outdir = pathlib.Path('./')
er.write(outdir=outdir, start_date=startdate, rnday=rnday, air=True, rad=True, prc=True, bbox=bbox, overwrite=True)print(f'It took {(time()-t0)/60} minutes to generate {rnday} days')

**Comment by SorooshMani-NOAA** [comment]
I see. I can update my client script to use ERA5 directly, but is there any plan for updating ERA5 to work like GFS and HRRR?

**Comment by SorooshMani-NOAA** [comment]
@cuill, writing ERA5 alone doesn't solve the issue completely. I also need to manually modify `model_driver_obj.param.opt.wtiminc` and `model_driver_obj.param.opt.nws` since it won't be automatically updated based on wind forcing in `ModelDriver.__init__` at https://github.com/schism-dev/pyschism/blob/ec00dffb251e49759a1162eff4e911c82a7d0cf2/pyschism/driver.py#L348-L355

So I think it'd be good idea to make the API similar to what NWS2 superclass expects.
@jreniel can you please comment on this?

**Comment by jreniel** [comment]
This is what happens when we prioritize "make it fast" over "make it right". I've been aware of this issue for a long time and will eventually fix it. If you want to provide a fix, you are welcome to open a PR. Thanks.

**Comment by SorooshMani-NOAA** [comment]
I need to first familiarize myself with the nws module in pyschism. If I get to it I'll open a PR.

**Comment by SorooshMani-NOAA** [comment]
This might be obvious, but I'm just writing it for the sake of documentation:

I noticed that since `ERA5` is bypassing `NWS2`, other than updating relevant parameters, the user needs to write out `sflux_inputs.txt` and `windrot_geo2proj.gr3` manually as well. In my case I'm just writing an empty namelist file for `sflux_inputs.txt` and also creating and writing `Windrot` to the SCHISM input dir (following the logic from `NWS2.write`).

**Comment by cuill** [comment]
@huyquangtranaus Yes, it generates 3-hourly netCDF files. Originally, it was designed to satisfy our group's needs. I think it's better to add the output interval as a parameter to let the user decide. I will consider adding this feature. Thanks.

**Comment by huyquangtranaus** [comment]
@cuill Yeah... that is useful too. Thank you.

Also, there is likely  an issue in the subset forcing area is in the southern hemisphere (SH) because you use ::-1 to reverse the list, for [example]
(https://github.com/schism-dev/pyschism/blob/d2b474e8a32e22eea28f7da462feae8424ab0ca4/pyschism/forcing/nws/nws2/era5.py#L131)

So the 2d matrix is flipped like this:  
![Screenshot from 2022-05-27 20-19-52](https://user-images.githubusercontent.com/32722297/170682202-d56458c2-2e11-4558-9971-a3c2394ef8af.png)

Have you seen this?
